Netkan/Processors/QueueHandler.cs

```diff
@@ -0,0 +1,206 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Amazon;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using log4net;
+using CKAN.NetKAN.Model;
+
+namespace CKAN.NetKAN.Processors
+{
+    public class QueueHandler
+    {
+        public QueueHandler(string cacheDir, bool overwriteCache, string githubToken, bool prerelease)
+        {
+            log.Debug("Initializing SQS queue handler");
+            // TODO: Shouldn't hard code the region
+            client   = new AmazonSQSClient(RegionEndpoint.USWest2);
+            inflator = new Inflator(cacheDir, overwriteCache, githubToken, prerelease);
+
+            inputQueueURL  = getQueueUrl(inputQueueName);
+            outputQueueURL = getQueueUrl(outputQueueName);
+            log.DebugFormat("Queue URLs: {0}, {1}", inputQueueURL, outputQueueURL);
+        }
+
+        public void Process()
+        {
+            while (true)
```

Comment:
From my perspective, this is a GoodEnough&trade; starting point. If you're interested you could do something with signal handlers, so that we can bail out gracefully when a term signal is sent. But that's also rather linux specific, I don't know how cross platform the logic is (though we only care about linux here).

Reply:
OK, as long as it won't be a problem for managing the container. I know it's easy enough to kill a process, but if there's a cleaner way to shut down, just let me know.